lib/publiccloud/instance.pm

```diff
@@ -616,12 +616,25 @@ sub measure_boottime() {
     $ret->{kernel_release} = $instance->run_ssh_command(cmd => 'uname -r', proceed_on_failure => 1);
     $ret->{kernel_version} = $instance->run_ssh_command(cmd => 'uname -v', proceed_on_failure => 1);
 
-    # Do logging to openqa UI
     $Data::Dumper::Sortkeys = 1;
+    my $dir = "/var/log";
+    my @logs;
+    my @missing_logs;
+
+    # Select existing files
+    ($instance->run_ssh_command(cmd => "test -e $dir/" . $_, timeout => 0, proceed_on_failure => 1) == 0) ? push(@logs, $_) : push(@missing_logs, $_)
+      foreach (qw(cloudregister cloud-init.log cloud-init-output.log messages NetworkManager));
```

Comment:
I fear this is getting out of control and a bit unreadable, imho.  The use of the ternary operator should be reserved for assignments and not control flow, which is the realm of if/else.

```suggestion
    foreach (qw(cloudregister cloud-init.log cloud-init-output.log messages NetworkManager)) {
        if ($instance->run_ssh_command(cmd => "test -e $dir/" . $_, timeout => 0, proceed_on_failure => 1) == 0) { 
            push(@logs, $_);
        } else {
            push(@missing_logs, $_);
        }
```

Reply:
It was a suggestion of Anton and I agreed it is valid, to me synthesis is welcome, because we're doing log-upload, not a critical functionality, moreover it helps to soon recognize the focal changes.

Moreover IMHO spreading `if` and `{ ...}` fills the routine of equivalent function but with more glittering.

But, as you can see, I added `WIP` everywhere, because I need to change something more 'cause of different output formats of run_ssh_commands in my checks. 
Review back soon :slightly_smiling_face:

Reply:
The ternary operator is fine for assignments but _not_ control flow.  That's the job of if/else.

The current code is cluttering in the horizontal axis.  Using if/else doesn't add any noise in the vertical axis.  And it's actually more readable.

The reason the ternary operator wasn't included in Golang and  Python it's this kind of abuse... =/

Reply:
I see: it is a question of readability, but i.e. for me it is more readable this way; from a machine p.o.v. it is equivalent or could be even a bit faster.

Consider that originally I wanted to move all that logs code outside, in a separated routine, leaving here a simple call or much less code: moving to that perspective, I'd agree with you, because synthesis achieved already with the routine call...

Reply:
@asmorodskyi can you please go back in time and backtrack your suggestion? :)

Reply:
if @ricardobranco777  will give +1 to your initial version which I declined I would accept it

Reply:
But still I like much more _ternary_ option :smile: :smile: :smile:

Reply:
> if @ricardobranco777 will give +1 to your initial version which I declined I would accept it

I agreed to the change of name.

Reply:
> But still I like much more _ternary_ option, that awakened my style ... 😄 😄 😄

Ternary operators are fine for assignments like in `my $VARIABLE = CONDITION ? EXPRESSION_1 : EXPRESSION_2;` but you're using it like `CONDITION ? STATEMENT_1 : STATEMENT_2;`, which is a psychedelic abomination I didn't even know was possible. 

Imagine changing every if/else for ternary operators...  That's why this operator didn't make it to Python, Go & Rust.

A cool way to simplify the code would be to drop the whole `@missing_logs` thing as we can infer that logs are missing in the `Logs & Assets` tab.  That would make it indeed faster as it avoids running ssh commands for each file.

Reply:
I'd not "always" use and don't everywhere do use it: in This specific case I preferred this kind of fix and neither regressions nor other issues have been introduced here.

Please, a kind look of last format of the solution fix: it appear simple, clear, synthetic and functional working; VR are successful.

Reply:
> IMHO such words are excessive and should be never part of a review, we here are back in a discussion of style, one can agree or not, 

It's just a tongue-in-cheek expression meant to be taken lightly.

> but where fix is provided and no other issue added. I see neither computers nor code claiming any help and salvation from a dummy programmer or a blind typing monkey because of this ternary use: those are still working well 🙂. But still may be I am wrong, and if you address me to some guideline link related to your topic, we can discuss of it.

I do have a strong opinion in this case because it's not a matter of personal taste or style.  This is _bad_ coding practice.  Ternary operators are not meant for control flow.  See how it's used only in different languages only for assignments:

- https://en.wikipedia.org/wiki/Ternary_conditional_operator
- https://therealadam.com/2024/03/28/avoid-the-ternary.html
- https://go.dev/doc/faq#Does_Go_have_a_ternary_form
- https://dev.to/chinesehuazhou/why-dont-python-go-and-rust-have-a-ternary-conditional-operator-3jk7

> Please, a kind look of last format of the solution fix: it appear simple, clear, synthetic and functional working; VR are successful.

What do you mean by synthetic?  Avoiding if/else?